roles/pulp_common/tasks/install_pip.yml

```diff
@@ -250,6 +242,13 @@
         clients: "{{ pulp_install_dir }}/bin/pip"
       register: pip_pkgs
 
+    - name: Install patched versions of dependencies (FIPS only)
+      shell: |
+        django_location=$({{ pulp_install_dir }}/bin/pip3 show Django | grep Location | cut -d' ' -f2)
+        sed -i 's/hashlib.md5()/hashlib.md5(usedforsecurity=False)/' "$django_location/django/db/backends/utils.py"
+      when: ansible_fips
```

Comment:
I propose we do two things:
1. Remove the `ansible_fips` variable and its detection entirely
2. We patch the dev enviornments regardless. This is useful because any dev environment can spot a problem with this instead of just the FIPS environments.

Reply:
How do we patch the dev envs? If we patch them in the pulp_dev role, the server will fail to start (in a fips environment) while the common part of the installer is running, causing ansible to stop midflight.

Reply:
We use a role check like this PR is already doing. Change the when condition to 'pulp_devel' in role_names.